client/src/backend/remote_node.rs

```diff
@@ -143,11 +141,11 @@ impl RemoteNode {
             Decode::decode(&mut &events_data[..]).map_err(Error::Codec)?;
 
         let opt_signed_block = self.rpc.chain.block(Some(block_hash)).compat().await?;
-        let block = opt_signed_block
-            .ok_or_else(|| {
-                Error::from("Block that should include submitted transaction does not exist")
-            })?
-            .block;
+        let opt_block =
+            opt_signed_block.map(|signed_block| unwrap_transport_block(signed_block.block));
+        let block = opt_block.ok_or_else(|| {
+            Error::from("Block that should include submitted transaction does not exist")
+        })?;
```

Comment:
Maybe `unwrap_transport_block` could be renamed to `signed_block_to_block` and accept a signed block? Then this could clear up:
```suggestion
        let block = opt_signed_block.map(signed_block_to_block)
            .ok_or("Block that should include submitted transaction does not exist")?;
```

Reply:
I’d like to keep `unpack_transport_block` because it makes the one-to-one-correspondence between `TransportBlock` and `Block` more obvious and it clearly separates the workaround for the serialization from the actual logic.

Reply:
That's fair